### DISCUSSION TITLE
`# pylance: basic` didn't work

### DISCUSSION THREAD

**Discussion by andyluooo** [microsoft/pylance-release#4394] [Q&A] [answered]
I want to change the type checking mode of a specific file to `basic` (default `strict` in the settings.json). I refered to the documentation of pyright and tried to use `# pylance: basic` at top of my file similarly, which however didn't work. What should I do?

**Comment by erictraut** [comment]
I'm not able to repro what you're seeing. I changed "python.analysis.typeCheckingMode" to "strict" in my settings, then I created a new file in the project and created an error that is reported only in strict mode. Then I added `# pyright: basic` to the top of that file, and the error went away. Are you seeing something different?

**Reply by debonte** [reply, marked as answer]
> I refered to the documentation of pyright and tried to use # pylance: basic at top of my file similarly

I think we don't support `# pylance` comments?

**Reply by andyluooo** [reply]
> > I refered to the documentation of pyright and tried to use # pylance: basic at top of my file similarly
> 
> I think we don't support `# pylance` comments?

Thanks. It seems that I mixed up something. `# pyright: basic` works for me.

**Reply by erictraut** [reply]
Oh, good catch. I didn't notice that the poster used `# pylance`. That explains the problem. You need to use `# pyright`. (The name of the type checker is pyright. Pylance is the language server built on top of pyright.)

For more details about controlling behaviors with comments, refer to [this documentation](https://microsoft.github.io/pyright/#/comments).
